broker/broker.go

```diff
@@ -232,10 +235,39 @@ func (b *Broker) Update(ctx context.Context, instanceID string, details brokerap
 		}
 	}
 
-	if userParameters.MaxMemoryPolicy == nil && userParameters.PreferredMaintenanceWindow == "" {
+	if userParameters.MaxMemoryPolicy == nil && userParameters.PreferredMaintenanceWindow == "" && userParameters.TestFailover == nil {
 		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("no parameters provided")
 	}
 
+	if userParameters.TestFailover != nil {
+
+		planConfig, err := b.config.GetPlanConfig(details.PlanID)
+		if err != nil {
+			return brokerapi.UpdateServiceSpec{}, errors.Wrap(err, "Failed to find service plan")
+		}
+		if planConfig.Parameters["cluster-enabled"] == "yes" {
+			return brokerapi.UpdateServiceSpec{}, fmt.Errorf("Test failover does not support for cluster mode")
+		}
+		if planConfig.ReplicasPerNodeGroup < 1 {
+			return brokerapi.UpdateServiceSpec{}, fmt.Errorf("Test failover requires one or more replicas")
+		}
+		if userParameters.MaxMemoryPolicy != nil || userParameters.PreferredMaintenanceWindow != "" {
+			return brokerapi.UpdateServiceSpec{}, fmt.Errorf("Test failover must be used by itself")
+		}
+		primaryNode, err := b.provider.TestFailover(providerCtx, instanceID)
+		if err != nil {
+			return brokerapi.UpdateServiceSpec{}, errors.Wrap(err, "Test failover failed: ")
+		}
+		return brokerapi.UpdateServiceSpec{
+			IsAsync: true,
+			OperationData: Operation{
+				Action:      ActionFailover,
+				PrimaryNode: primaryNode,
+				TimeOut:     time.Now().Add(45 * time.Minute).Format(time.RFC3339),
```

Comment:
Can we extract this to a var at the top of the file so it’s less of a magic value?